nearcore/src/config.rs

```diff
@@ -1293,15 +1294,16 @@ pub fn load_config(
             if let Err(e) = genesis.validate(genesis_validation) {
                 validation_errors.push_errors(e)
             };
-            if validator_signer.is_some()
+            // Before stateless validation we require validators to track all shards, see
+            // https://github.com/near/nearcore/issues/7388
+            if !checked_feature!("stable", StatelessValidationV0, PROTOCOL_VERSION)
+                && validator_signer.is_some()
                 && matches!(
                     genesis.config.chain_id.as_ref(),
                     near_primitives::chains::MAINNET | near_primitives::chains::TESTNET
```

Comment:
can you also remove this chain-id check? This should not be specific to mainnet or testnet.

Reply:
I believe this was used for production chains only so that we can still set `tracked_shards=[]` in localnet, mocknet, etc.

Reply:
do you know why we set it to empty for localnet/mocknet and what is the side effect of it? maybe to test what happens if the node tracks no shards?

I do not think removing the chain-id check is necessary for this PR, though, cleaning up the chain ids could be done separately.